modules/host/dependencies.go

```diff
@@ -0,0 +1,79 @@
+package host
+
+import (
+	"errors"
+	"net"
+	"os"
+
+	"github.com/NebulousLabs/Sia/persist"
+)
+
+// Fake errors that get returned when a simulated failure of a dependency is
+// desired for testing.
+var (
+	mockErrListen       = errors.New("simulated Listen failure")
+	mockErrLoadFile     = errors.New("simulated LoadFile failure")
+	mockErrMkdirAll     = errors.New("simulated MkdirAll failure")
+	mockErrNewLogger    = errors.New("simulated NewLogger failure")
+	mockErrOpenDatabase = errors.New("simulated OpenDatabase failure")
+)
+
+// These interfaces define the Host's dependencies. Mocking implementation
+// complexity can be reduced by defining each dependency as the minimium
+// possible subset of the real dependency.
+type (
+	// dependencies defines all of the dependencies of the Host.
+	dependencies interface {
+		// Listen gives the host the ability to receive incoming connections.
+		Listen(string, string) (net.Listener, error)
```

Comment:
`logClose` vs `bufferClose` if you run into namespace issues. Not sure how I feel about lower-casing, I guess it makes sense.